slack_global_report.py

```diff
@@ -0,0 +1,241 @@
+        self.fake_channel = channel.Channel(fake=True)
+        self.channel = channel.Channel(fake=fake)
+        self.rn = random_name.RandomName()
+        self.user = user.User(fake=fake)
+        self.client = slack.WebClient(token=slack_token.token)
+        self.enricher = enricher.Enricher(fake=fake)
+        self.cml = channel_members_log.ChannelMembersLog()
+        self.report_channel  = self.channel.get(config.report_channel)['slack_cid']
+
+    def make_header(self, ur, pur):
+        blocks = []
+        header = "*Slack Activity Report for the Week Between {} and {}*"
+        header = header.format(ur['start_date'], ur['end_date'])
+        blocks.append(self.sf.text_block(header))
+        stats = ur['statistics']
+        posters = int(stats['posters'])
+        active_users = stats['active_users']
+        percent = (posters * 100.0) / active_users
+        text = "*{:,}/{:,}* (or *{:.1f}%*) users posted messages\n".format(posters, active_users, percent)
+        text += "Median message count was *{}*\n".format(stats['median messages'])
+        text += "The top ten posters contributed *{:.1f}%* of all messages (lower is better)\n".format(stats['topten messages'])
+        text += "The top *{}* posters (higher is better) accounted for about 50% of total volume\n".format(stats['50percent of words'])
+        blocks.append(self.sf.text_block(text))
+        w = stats['words']
+        m = stats['messages']
+        # it = interim text
+        it = "People posted *{:,}* words in *{:,}* messages (approximately *{:,}* pages), "
+        it += "or about *{:.1f}* words per message, *{:.1f}* words per poster, "
+        it += "or *{:.1f}* messages per poster"
+        text = it.format(int(w), int(m), int(w / 500), w/m, w/posters, m/posters)
+        blocks.append(self.sf.text_block(text))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def top_channels(self, ur, pur):
+        blocks = []
+        top = 20
+        header = "*Top {} Channels*".format(top)
+        blocks.append(self.sf.text_block(header))
+        channels = ur['channels']
+        cids = list(channels.keys())[:top]
+        cinfo = ur['channel_info']
+        cstats = ur['channel_stats']
+        cusers = ur['channel_user']
+        for idx, channel in enumerate(cids):
+            it = "{}. {} ".format(idx + 1, self.sf.show_cid(channel))
+            ci = cinfo[channel]
+            cs = cstats[channel]
+            cu = cusers[channel]
+            if ci['new']:
+                it += " (new)"
+            it += "{:,} members ".format(ci['members'])
+
+            m = channels[channel][0]
+            w = channels[channel][1]
+            p = len(cu)
+            it += "*{:,}* posters *{:,}* words *{:,}* messages ".format(p, w, m)
+            it += "*{:.1f}* words/poster ".format(w/p)
+            it += "*{:.1f}%* of total traffic, ".format(cs['percent'])
+            it += "*{:.1f}%* cumulative of total ".format(cs['cpercent'])
```

Comment:
Before: "#cooking 426 members 66 posters 8,456 words 451 messages 128.1 words/poster 3.3% of total traffic, 16.0% cumulative of total"

Can you add commas so it looks like this?
"#cooking: 426 members, 66 posters, 8,456 words, 451 messages, 128.1 words/poster, 3.3% of total traffic, 16.0% cumulative of total"

Reply:
Done